### PR TITLE
Documented Console 1.1 Improvements.

### PR DESCRIPTION
**Pull request by cwanda**
* Added description on how to use the Mynewt 1.0 and 1.1 console API for reading from console.
* Added note about support for SEGGER RTT
* Added descriptions for  console_set_queues() and console_non_blocking_mode().
* Updated description and example of console_init()
* Updated description of console_is_init()
* Updated example for console_read()
* Updated description for console_blocking_mode()

Note that I did not document the console_set_completion_cb() function since I think only the shell uses it and we do not need to expose it to other users.

### CODE REVIEW

File: docs/os/modules/console/console.md [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+* `CONSOLE_UART_BAUD`: Specifies the UART baud rate. The default is 115200.
+* `CONSOLE_UART_FLOW_CONTROL`: Specifies the UART flow control. The default is `UART_FLOW_CONTROL_NONE`.
+* ` CONSOLE_UART_TX_BUF_SIZE`: Specifies the transmit buffer size and must be a power of 2. The default is 32.
+<br>
+
+The `CONSOLE_TICKS` setting enables the console to print the current OS ticks in each output line. 
+
+**Notes:** 
+
+* SEGGER RTT support is not available in the Mynewt 1.0 console package .
```
Comment (mkiiskila) [inline docs/os/modules/console/console.md]: Typo: there's a space before .

File: docs/os/modules/console/console_blocking_mode.md [hunk 2]
```diff
@@ -4,9 +4,7 @@
 void console_blocking_mode(void)
 ```
 
-Calls the `console_blocking_tx` function to flush the buffered console output (transmit) queue. 
-The function `OS_ENTER_CRITICAL()` is called to disable interrupts and `OS_EXIT_CRITICAL()` 
-is called to enable interrupts back again once the buffer is flushed.
+Sets the transmit mode to blocking. The function flushes the buffered console output.  The function `OS_ENTER_CRITICAL()` is called to disable interrupts and `OS_EXIT_CRITICAL()` is called to enable interrupts back again once the buffer is flushed.
```
Comment (mkiiskila) [inline docs/os/modules/console/console_blocking_mode.md]: User should have interrupts disabled when calling this, and should not enable them after the call. UART drivers do not adjust their interrupt delivery settings when console is in 'blocking TX mode', so system behaviour would be undefined/undeterministic, should the user re-enable interrupts.

File: docs/os/modules/console/console.md (hunk 1 above)
Comment (cwanda) [inline docs/os/modules/console/console.md]: Fixed. Thanks

File: docs/os/modules/console/console_blocking_mode.md (hunk 2 above)
Comment (cwanda) [inline docs/os/modules/console/console_blocking_mode.md]: I have removed the description for transmit mode  and the console_blocking_mode and console_non_blocking_mode functions.

**Comment by cwanda** [comment]
Close due to repo change to master

**Pull request closed without merging**